pandas/core/groupby/generic.py

```diff
@@ -963,16 +963,16 @@ def aggregate(self, func=None, *args, **kwargs):
                         [self._selected_obj.columns.name] * result.columns.nlevels
                     ).droplevel(-1)
 
-        if not self.as_index:
-            self._insert_inaxis_grouper_inplace(result)
-            result.index = np.arange(len(result))
-
         if relabeling:
 
             # used reordered index of columns
             result = result.iloc[:, order]
             result.columns = columns
 
+        if not self.as_index:
```

Comment:
Could you put the issue number as a comment above this line and briefly describe the issue

Reply:
Hi, I have added the comment.